mapbox_vector_tile/decoder.py

```diff
@@ -89,9 +90,18 @@ def parse_geometry(self, geom, ftype):
             i = i + 1
 
             if cmd == CMD_SEG_END:
-                break
+                if ftype == POLYGON and coords:
+                    coords.append(coords[0])
+                parts.append(coords)
+                coords = []
+
+            elif cmd == CMD_MOVE_TO or cmd == CMD_LINE_TO:
+
+                if ftype == LINESTRING and coords and cmd == CMD_MOVE_TO:
```

Comment:
Does this need to be `ftype == POLYGON` as well? The spec says that `CMD_MOVE_TO` also signals the end of a ring, and I couldn't figure out where else that might be happening...